include/fmt/format-inl.h

```diff
@@ -531,14 +532,14 @@ class bigint {
 
   friend struct formatter<bigint>;
 
-  void subtract_bigits(int index, bigit other, bigit& borrow) {
+  void subtract_bigits(size_t index, bigit other, bigit& borrow) {
```

Comment:
In all the bigint and floating-point formatting code let's keep the type of indices `int`. More generally we don't need to address false positives in `format-inl.h` because it's an internal header.

Reply:
The code is trying to subscript into bigits object (of type buffer) with a signed value, when the operator [] expects std::size_t, thus the warning.

Do you want me to convert the buffer implementation and interface from size_t to int? This, after all, what Scott Meyers and Bjarne Stroustroup are recommending:
     http://open-std.org/JTC1/SC22/WG21/docs/papers/2019/p1428r0.pdf
     https://www.aristeia.com/Papers/C++ReportColumns/sep95.pdf

Converting to int from size_t *should* clean up a lot of casts.

Reply:
Changing `buffer`'s index to a signed type might be cleaner but let's not do this in the current PR. For now let's just keep floating-point related code as is.

Reply:
OK, will leave changing buffer for a follow-up change.